src/main/java/org/reactome/release/qa/check/CoVEntityDiseaseCheck.java

```diff
@@ -0,0 +1,88 @@
+package org.reactome.release.qa.check;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.reactome.release.qa.annotations.SliceQACheck;
+import org.reactome.release.qa.common.AbstractQACheck;
+import org.reactome.release.qa.common.QACheckerHelper;
+import org.reactome.release.qa.common.QAReport;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This QA check was mostly used during the CoV-1-to-CoV-2 inference process. For now it is being kept, but may
+ * be removed, given it was for a specific curations (August 2020).
+ *
+ * This checks for instances that have either a CoV-1 or CoV-2 species without the corresponding disease attribute.
+ * For CoV-1 instances, they would be missing 'severe acute respiratory disease' (9678120). For CoV-2 instances, they
+ * would be missing 'COVID-19' (9683912).
+ *
+ * @author jcook
+ */
+
+@SliceQACheck
+public class CoVEntityDiseaseCheck extends AbstractQACheck {
+
+    @Override
+    public QAReport executeQACheck() throws Exception {
+        QAReport report = new QAReport();
+
+        // Fetches all Events and PhysicalEntities in DB
+        Collection<GKInstance> eventsAndPhysicalEntities = new ArrayList<>();
+        eventsAndPhysicalEntities.addAll(dba.fetchInstancesByClass(ReactomeJavaConstants.Event));
+        eventsAndPhysicalEntities.addAll(dba.fetchInstancesByClass(ReactomeJavaConstants.PhysicalEntity));
```

Comment:
Is it necessary to check ALL Events and PEs? Why not filter for Events and PEs with species=CoV-1 or species=Cov-2 and _then_ loop over those, checking for the disease attribute?

Reply:
I think it came down to it being a bit complicated to search for: You want to search `species` and `relatedSpecies` attributes, which can both be multi-value, with 'instance' values. I think I've found that searching by attributes where the value would be an instance wasn't really working using the java API

Reply:
Ah, I didn't realise that both species attributes are multi-valued. I think there _is_ a way to do that with the Reactome API, but it's a bit tricky. Probably possible to write a query in SQL that returns just a list of DB_IDs and then iterate on that to fetch by DB_ID and then perform the other checks... I usually like to leave such things as filtering to the database, because it's _good_ at that (sort of what it was made for), but if this already works and isn't too slow and the API makes it too hard to do it, then you might as well leave it... ;)

Reply:
To @SolomonShorser-OICR's point, you could restructure the method body like this:

```
allEventsAndPhysicalEntities
.stream()
.filter(hasCov1SpeciesWithoutCov1Disease() || hasCov2SpeciesWithoutCov2Disease())
.forEach(covInstance -> report.addLine(getReportLine(covInstance));

report.setColumnHeaders(getColumnHeaders());
return report;
```
creating helper methods called in the filter to abstract out the details of checking species and disease tags.

The `getReportLine` function can then be expanded a little bit to determine which issue message should be used:

```
private String getReportLine(GKInstance covInstance) {
    String issue = hasCov1Species(covInstance) ? "COV-1 species without severe acute respiratory syndrome disease" : "COV-2 species without COVID-19 disease";

    return String.join("\t",
        instance.getDBID().toString(),
        instance.getDisplayName(),
        instance.getSchemClass().getName(),
        QACheckerHelper.getLastModificationAuthor(instance),
        issue
    );
}
```

Alternatively, you could keep the for loop and create two helper methods - one for getting the species db ids and the other for getting the disease db ids both with the instance passed in as a parameter.

Reply:
So I ended up going with a bit of a hybrid of @jweiser's suggestions:
- I kept the for loop, as my code style throughout the Reactome project doesn't use streams. I'm not exactly convinced they help with code clarity, despite the fact they are handy ways of writing a lot into a small amount of space
- I did my best to turn the code blocks into discrete methods that describe exactly what they do
- I used the ternary suggestion in the `getReportLine` method